test/testutils.jl

```diff
@@ -0,0 +1,330 @@
+
+struct ImputorTester{I<:Imputor}
+    imp::Type{I}
+    f::Function
+    f!::Function
+    kwargs::Dict
+end
+
+function ImputorTester(imp::Type{<:Imputor}; kwargs...)
+    fname = lowercase(string(nameof(imp)))
+    settings = Dict(kwargs...)
```

Comment:
Why are these mutable? Why not just make the Kwargs be `settings` and leave it in its named tuple?

Reply:
FWIW, kwargs isn't a namedtuple, but yes, a namedtuple would work better.